x-pack/solutions/security/plugins/security_solution/common/detection_engine/prebuilt_rules/diff/extract_threat_array.ts

```diff
@@ -8,21 +8,43 @@
 import type {
   RuleResponse,
   ThreatArray,
+  ThreatSubtechnique,
   ThreatTechnique,
 } from '../../../api/detection_engine/model/rule_schema';
 
 export const extractThreatArray = (rule: RuleResponse): ThreatArray =>
   rule.threat.map((threat) => {
     if (threat.technique && threat.technique.length) {
-      return { ...threat, technique: trimTechniqueArray(threat.technique) };
+      return {
+        ...threat,
+        tactic: { ...threat.tactic, reference: normalizeThreatReference(threat.tactic.reference) },
+        technique: trimTechniqueArray(threat.technique),
+      };
     }
-    return { ...threat, technique: undefined }; // If `technique` is an empty array, remove the field from the `threat` object
+    return {
+      ...threat,
+      tactic: { ...threat.tactic, reference: normalizeThreatReference(threat.tactic.reference) },
+      technique: undefined,
+    }; // If `technique` is an empty array, remove the field from the `threat` object
   });
 
 const trimTechniqueArray = (techniqueArray: ThreatTechnique[]): ThreatTechnique[] => {
   return techniqueArray.map((technique) => ({
     ...technique,
+    reference: normalizeThreatReference(technique.reference),
     subtechnique:
-      technique.subtechnique && technique.subtechnique.length ? technique.subtechnique : undefined, // If `subtechnique` is an empty array, remove the field from the `technique` object
+      technique.subtechnique && technique.subtechnique.length
+        ? trimSubtechniqueArray(technique.subtechnique)
+        : undefined, // If `subtechnique` is an empty array, remove the field from the `technique` object
   }));
 };
+
+const trimSubtechniqueArray = (subtechniqueArray: ThreatSubtechnique[]): ThreatSubtechnique[] => {
+  return subtechniqueArray.map((subtechnique) => ({
+    ...subtechnique,
+    reference: normalizeThreatReference(subtechnique.reference),
+  }));
+};
+
+const normalizeThreatReference = (reference: string): string =>
+  !reference.endsWith('/') ? `${reference}/` : reference; // Adds a trailing backslash in urls if it doesn't exist to account for any inconsitencies between our script generated data and prebuilt rules packages
```

Comment:
The current implementation won't work in general case. For example `reference` may have query string added in the future. Let's make the implementation more robust

```ts
const normalizeThreatReference = (reference: string): string => {
  try {
    const parsed = new URL(reference);

    if (!parsed.pathname.endsWith('/')) {
      // Adds a trailing backslash in urls if it doesn't exist to account for
      // any inconsistencies between our script generated data and prebuilt rules packages
      parsed.pathname = `${parsed.pathname}/`;
    }

    return parsed.toString();
  } catch {
    return reference;
  }
};
```